Library/Homebrew/utils/curl.rb

```diff
@@ -118,36 +118,40 @@ def curl(*args, print_stdout: true, **options)
       result
     end
 
-    def curl_download(*args, to: nil, partial: true, **options)
+    def parse_headers(headers)
+      return {} unless headers
+
+      headers.split("\n").to_h do |h|
+        partitioned = h.partition(": ")
+        [partitioned.first, partitioned.last]
```

Comment:
I may have misunderstood but wouldn't this be equivalent:
```suggestion
        h.partition(": ")
```

Reply:
The array that `partition` returns contains the separator. `split` would be a better choice here, so I've made the change.